.ci/jenkins_job_build.sh

```diff
@@ -210,6 +210,12 @@ if [ "${METRICS_CI}" = "false" ]; then
 		# If not a PR, we are testing on stable or master branch.
 		[ -z "$pr_number" ] && specific_branch="true"
 		"${ci_dir_name}/static-checks.sh" --only-arch "$kata_repo" "$specific_branch"
+		# It is too early to configure defaultRuntimeConfiguration for CC
+		# The file generation should be deferred at runtime installation with
+		# /opt/confidential-containers/share/defaults/kata-containers/configuration.toml
+		if [ "$KATA_BUILD_CC" == "yes" ]; then
```

Comment:
This isn't necessarily a suggestion, but I wanted to open the discussion to see what people think:
- Is there an argument that we remove the conditional and always delete (or attempt to) the `src/runtime/pkg/katautils/config-settings.go` after the static-checks. At the moment it only impacts CC builds, but it was a bit of a bear trap and difficulty debugging, so I don't want someone else to hit that incase there is another scenario that we change the default config location. As this doesn't impact x86 jobs it's more easy to over look.

Another left field suggestion is how far are we away from being able to run the s390x static checks on GHA in which case we can remove the early creation of this?

Reply:
Here are my two cents even if I have little knowledge why this code block is placed in this way:

1. I agree with removing the condition if it does not break the existing CI jobs.
2. Regarding the static checks, I could make it separate and run it on GHA for s390x (putting in my backlog :wink:), but not sure if we could remove this after it. I am wondering how other architectures are planning to introduce a GHA CI job for this.

cc: @Amulyam24 @ryansavino

Reply:
@stevenhorsman, this will hit the other CIs by the moment we start using the kata-deploy scripts as part of the `main` branch.  So, yep, I'd be in favour of removing the file after running the static-checks (as the first step).

Then, a second step would be to use the GHA to run the static checks for all the arches.  AFAIR GitHub does provide ARM machines (so, no need to have an external runner), and we can rely on the external s390x one.  That'd be my dream. :-)

Reply:
Hey Choi - sorry to un-resolve this discussion, but I'm not sure we got to a conclusion, unless I've missed something. You stated:
> I agree with removing the conditional if it does not break the existing CI jobs.

Have we been able to test that as my (biased) interpretation is that this approach is supported by all three of us?

Reply:
> this will hit the other CIs by the moment we start using the kata-deploy scripts as part of the main branch. 

I thought that the conditional is needed for other CIs, isn't it? Otherwise, we could try another run without it. Sorry, I am not good at understanding something clearly. :wink:

Reply:
Ok, sorry - it is as likely that I'm misinterpreting, so I'll be clearer here:
Can we remove the `if [ "$KATA_BUILD_CC" == "yes" ]; then` condition from this PR as I think that will resolve the issue for all the CIs just here rather than deferring it until we hit it later? I don't see a down side to this, but obviously we'll need to re-run the tests to double check.

Reply:
Just seen your re-push. Thanks!